### PR TITLE
Add support of Falcon-H1 models for liger kernels

### PR DESCRIPTION
**Pull request by puneeshkhanna**
## Summary
This PR enables ROPE, rmsnorm and cross entropy liger kernels for Falcon-H1 models 

Falcon H1 models - 
https://huggingface.co/collections/tiiuae/falcon-h1-6819f2795bc406da60fab8df

Support for fused cross entropy and swiglu to be added later.

## Testing Done
Verified fine tuning Falcon H1 models with and without enabling liger kernels and loss plot matches. And gives speedup as well as memory savings.

- Hardware Type: H100
- [ ] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [ ] run `make test-convergence` to ensure convergence

### CODE REVIEW

File: src/liger_kernel/transformers/monkey_patch.py [hunk 1]
```diff
@@ -1956,6 +2031,7 @@ def apply_liger_kernel_to_glm4v(
     "smollm3": apply_liger_kernel_to_smollm3,
     "phi3": apply_liger_kernel_to_phi3,
     "paligemma": apply_liger_kernel_to_paligemma,
+    "falcon_h1": apply_liger_kernel_to_falcon_h1,
```
Comment (shimizust) [inline src/liger_kernel/transformers/monkey_patch.py:2209]: Can you add this to src/liger_kernel/transformers/__init__.py?

File: src/liger_kernel/transformers/monkey_patch.py [hunk 2]
```diff
@@ -1928,6 +1928,81 @@ def apply_liger_kernel_to_glm4v(
                     _patch_rms_norm_module(decoder_layer.post_mlp_layernorm)
 
 
+def apply_liger_kernel_to_falcon_h1(
```
Comment (shimizust) [inline src/liger_kernel/transformers/monkey_patch.py:1931]: Can you add corresponding convergence and monkey patch tests?

File: src/liger_kernel/transformers/monkey_patch.py [hunk 3]
```diff
@@ -1928,6 +1928,81 @@ def apply_liger_kernel_to_glm4v(
...
+    Args:
+        rope (bool): Whether to apply Liger's rotary position embedding. Default is True.
+        cross_entropy (bool): Whether to apply Liger's cross entropy loss. Default is False.
+        fused_linear_cross_entropy (bool):
+            Whether to apply Liger's fused linear cross entropy loss. Default is True.
+            `cross_entropy` and `fused_linear_cross_entropy` cannot both be True.
+            If `fused_linear_cross_entropy` is True, the logits will not be materialized but more memory efficient.
+        rms_norm (bool): Whether to apply Liger's RMSNorm. Default is True.
+        swiglu (bool): Whether to apply Liger's SwiGLU MLP. Default is True.
+        model (PreTrainedModel): The model instance to apply Liger kernels to, if the model has already been
```
Comment (shimizust) [inline src/liger_kernel/transformers/monkey_patch.py]: Set correct defaults for flce and swiglu if not implemented

File: src/liger_kernel/transformers/monkey_patch.py [hunk 4]
```diff
@@ -1928,6 +1928,81 @@ def apply_liger_kernel_to_glm4v(
...
+            modeling_falcon_h1.CrossEntropyLoss = LigerCrossEntropyLoss
+
+    # TODO: To be enabled
+    if fused_linear_cross_entropy:
+        if transformer_version >= version.parse(SUPPORTED_TRANSFORMER_VERSION):
+            modeling_falcon_h1.FalconH1ForCausalLM.forward = llama_lce_forward
+        else:  # if version < 4.46.1
+            logger.warning(TRANSFORMER_DEPRECATION_WARNING)
+            modeling_falcon_h1.FalconH1ForCausalLM.forward = llama_lce_forward_deprecated
+
```
Comment (shimizust) [inline src/liger_kernel/transformers/monkey_patch.py:1982]: Can you raise NotImplementedError in the case of swiglu and flce are set to true? Right now we're patching with incorrect modules (also in the case below where model instance exists)

**Review by shimizust — COMMENTED** [review]
If can't implement swiglu/flce in this impl, can you create a new issue to track this?

**Comment by puneeshkhanna** [comment]
Thanks for all the comments @shimizust . Will address soon
